File: checker-framework-manual-examples/11_5/scala/checker_framework_11_5.scala
import scala.util.boundary

def lessThan(arr1: Array[Int], arr2: Array[Int]): Boolean = boundary {
    for (i <- arr1.indices) {
        if (arr1(i) < arr2(i)) {
            boundary.break(true);
        } else if (arr1(i) > arr2(i)) {
            boundary.break(false)
        }
    }
    false
}

File: checker-framework-manual-examples/11_5/kotlin/checker_framework_11_5.kt

fun lessThan(arr1: Array<Double>, arr2: Array<Double>): Boolean {
    for (i in arr1.indices) {
        if (arr1[i] < arr2[i]) {
            return true;
        } else if (arr1[i] > arr2[i]) {
            return false;
        }
    }
    return false
}
